Documentation/project-docs/developer-guide.md

```diff
@@ -96,67 +112,61 @@ build-managed -debug -buildArch=x64
+For example the src\System.Diagnostics.DiagnosticSource directory holds the source code for the System.Diagnostics.DiagnosticSource.dll assembly. Each of these directories includes two projects, one for the DLL being built and one for the tests.
 
-You can build the tests for System.Diagnostics.DiagnosticSource.dll by going to 
-src\System.Diagnostics.DiagnosticSource\tests and typing `msbuild System.Diagnostics.DiagnosticSource.Tests.builds`.
+You can build the DLL for System.Diagnostics.DiagnosticSource.dll by going to the src\System.Diagnostics.DiagnosticsSource\src directory and typing `msbuild System.Diagnostics.DiagnosticSource.csproj`. The DLL ends up in bin\AnyOS.AnyCPU.Debug\System.Diagnostics.DiagnosticSource as well as bin\runtime\[BuildConfiguration].
+
+You can build the tests for System.Diagnostics.DiagnosticSource.dll by going to
+src\System.Diagnostics.DiagnosticSource\tests and typing `msbuild System.Diagnostics.DiagnosticSource.Tests.csproj`.
 
 There is also a pkg directory for each project, and if you go into it and type `msbuild`, it will build the DLL (if needed)
 and then also build the NuGet package for it. The NuGet package ends up in the bin\packages directory.
 
-For libraries that have multiple build configurations, the `.builds` file will have multiple binary outputs (one for each configuration). It is also possible to directly build the `.csproj` project in order to produce only a single configuration. Looking at the `.builds` file will tell you which configuration combinations are valid. For example, a builds file with these entries will have two valid combinations:
-```XML
-<Project Include="System.Net.NetworkInformation.csproj">
-  <OSGroup>Linux</OSGroup>
-</Project>
-```
-`msbuild System.Net.NetworkInformation.csproj /p:OSGroup=Linux`
+For libraries that have multiple build configurations they will have them listed in the `<BuildConfigurations>` property group, commonly found in a configurations.props file next to the csproj. When building the csproj the best configuratin in that configuration list will be choosen and set for the build.
 
-```XML
-<Project Include="System.Net.NetworkInformation.csproj">
-  <OSGroup>Windows_NT</OSGroup>
-  <TargetGroup>uap101</TargetGroup>
-</Project>
-```
-`msbuild System.Net.NetworkInformation.csproj /p:OSGroup=Windows_NT /p:TargetGroup=uap101`
+`msbuild System.Net.NetworkInformation.csproj /p:OSGroup=Windows_NT /p:TargetGroup=uap`
 
 **Note:** If building in a non-Windows environment, call `<repo-root>/Tools/msbuild.sh` instead of just `msbuild`.
 
 ### Building other OSes
 
 By default, building from the root will only build the libraries for the OS you are running on. One can
-build for another OS by specifying `build-managed -FilterToOSGroup=[value]` or build for all by specifying `build-managed -BuildAllOSGroups`.
+build for another OS by specifying `build-managed -os=[value]`.
```

Comment:
Pre-existing: Why does it talk only about build-managed and not either build or also about build-native?